contribs/gmf/src/services/treemanager.js

```diff
@@ -267,20 +283,77 @@ gmf.TreeManager.prototype.addGroupByName = function(groupName, opt_add) {
  * @param{string} layerName Name of the layer inside the group to add.
  * @param{boolean=} opt_add if true, force to use the 'add' mode this time.
  * @param{boolean=} opt_silent if true notifyCantAddGroups_ is not called
+ * @param{ol.Map=} opt_map Map object.
  * @export
  */
-gmf.TreeManager.prototype.addGroupByLayerName = function(layerName, opt_add, opt_silent) {
+gmf.TreeManager.prototype.addGroupByLayerName = function(layerName, opt_add, opt_silent, opt_map) {
   this.gmfThemes_.getThemesObject().then(function(themes) {
     var group = gmf.Themes.findGroupByLayerName(themes, layerName);
     if (group) {
-      this.addGroups([group], opt_add, opt_silent);
-      // FIXME: set the layer visible
+      var groupAdded = this.addGroups([group], opt_add, opt_silent);
+      if (opt_map) {
+        this.$timeout_(function() {
```

Comment:
It's required since the OL Layer is actually created after a new node (group) is added to the layertree and a LayerTreeController is instanciated. This happens in a digest loop so it's not synchronous.
We unfortunately need the OL layer to be create in `setLayerVisible_`. It would be much better if it would be possible to set the state in each tree node, but we can't.